### PR TITLE
Add the probot check-group action

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

See title

Part of https://github.com/Lightning-AI/lightning/pull/14294

Next steps:
1. ~~Make it run on `pull_request_target`~~ edit: not possible because of https://github.com/probot/probot/issues/1635. Means that this might not work in forks
2. Check functionality on different PRs
3. Unmark all jobs as required but check-group, WIP, pre-commit, check-schema
4. Merge https://github.com/Lightning-AI/lightning/pull/14294
5. Tweak checkgroup.yml and path filters as necessary
6. Profit

### Does your PR introduce any breaking changes? If yes, please list them.

None

### CODE REVIEW

**Review by akihironitta — APPROVED** [review]

File: .github/workflows/probot-check-group.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  required-jobs:
+    runs-on: ubuntu-latest
+    # if this timeout triggers, then the job needs to be manually restarted through the GitHub interface
+    timeout-minutes: 60
+    steps:
+      - uses: carmocca/probot@v2
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          interval: 180  # seconds
```
[the remark this answers is not in the document]
Comment (Borda) [reply .github/workflows/probot-check-group.yml:16]: well, if it takes up to 60min, what is the max of jobs that could be active at any moment? remember that we have only 160 agents for all our ptojects

[the remark this answers is not in the document]
Comment (awaelchli) [reply .github/workflows/probot-check-group.yml:16]: Aren't there other jobs that contribute to the API call count here, I mean jobs other than this probot?

**Review by awaelchli — APPROVED** [review]

File: .github/workflows/probot-check-group.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (carmocca) [reply .github/workflows/probot-check-group.yml:16]: >  what is the max of jobs that could be active at any moment?

As I said, the above estimation is for 10 jobs tops running at the same time.

> Aren't there other jobs that contribute to the API call count here, I mean jobs other than this probot?

Yes of course, this is why I chose a safety factor of 5 times the expected use. We can always decrease the timeout from 60s or increase the interval between checks.

File: .github/workflows/probot-check-group.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+name: Probot
+
+on:
+  pull_request:
+
```
Comment (Borda) [inline .github/workflows/probot-check-group.yml:5]:
```suggestion

concurrency:
  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.head_ref }}
  cancel-in-progress: true
  
```
to limit only one per PR

**Comment by carmocca** [comment]
I'm going to merge it to start seeing how it acts in different PRs. Might need tweaking. Most importantly, I need to check whether it will work on forks...

**Comment by carmocca** [comment]
> I need to check whether it will work on forks...

Looks like it's working in https://github.com/Lightning-AI/lightning/pull/14711

**Pull request merged**